lib/components/pagination.vue

```diff
@@ -10,290 +10,404 @@
+    data() {
+        return {
+            showFirstDots: false,
+            showLastDots: false,
+            currentPage: this.value
+        };
+    },
+    computed: {
+        numberOfPages() {
+            const result = Math.ceil(this.totalRows / this.perPage);
+            return (result < 1) ? 1 : result;
         },
-        computed: {
-            numberOfPages() {
-                const result = Math.ceil(this.totalRows / this.perPage);
-                return (result < 1) ? 1 : result;
-            },
-            btnSize() {
-                return this.size ? `pagination-${this.size}` : '';
-            },
-            pageLinks() {
-                if (this.currentPage > this.numberOfPages) {
-                    this.currentPage = 1;
-                }
+        btnSize() {
+            return this.size ? `pagination-${this.size}` : '';
+        },
+        pageList() {
+            // Sanity checks
+            if (this.currentPage > this.numberOfPages) {
+              this.currentPage = this.numberOfPages;
+            } else if (this.currentPage < 1) {
+              this.currentPage = 1;
+            }
 
-                // Defaults
-                this.diff = 1;
-                this.showPrev = false;
-                this.showNext = false;
+            // - Hide first ellipsis marker
+            this.showFirstDots = false;
+            // - Hide last ellipsis marker
+            this.showLastDots = false;
 
-                // If less pages than limit just show this pages
-                if (this.numberOfPages <= this.limit) {
-                    return this.numberOfPages;
-                }
+            let numLinks = this.limit;
+            let startNum = 1;
 
-                // If at the beginning of the list
-                if (this.currentPage <= this.limit - 2) {
-                    this.diff = 1;
-                    this.showNext = true;
-                    return this.limit - 2;
-                }
+            if (this.numberOfPages <= this.limit) {
+              // Special Case: Less pages available than the limit of displayed pages
+              numLinks = this.numberOfPages;
+            } else if (this.currentPage < (this.limit - 1) && this.limit > ELLIPSIS_THRESHOLD) {
+              // We are near the beginning of the page list
+              numLinks = numLinks - 1;
+              this.showLastDots = true;
+            } else if ((this.numberOfPages - this.currentPage + 2) < this.limit && this.limit > ELLIPSIS_THRESHOLD) {
+              // We are near the end of the list
+              this.showFirstDots = true;
+              numLinks = numLinks - 1;
+              startNum = this.numberOfPages - numLinks + 1;
+            } else {
+              // We are somewhere in the middle of the page list
+              if (this.limit > ELLIPSIS_THRESHOLD) {
+                this.showFirstDots = true;
+                this.showLastDots = true;
+                numLinks = numLinks - 2;
```

Comment:
`numLinks`, currently, is only mutated at most one time, as the ifs are all else ifs.  But yeah, it might be safer to set that as `numLinks = this.limit -2`